--- /home/dan/Desktop/ConjugateGradientReliableUpdate.h	2017-12-27 09:31:43.999037085 -0500
+++ lib/algorithms/iterative/ConjugateGradientReliableUpdate.h	2017-12-27 09:31:49.399776026 -0500
@@ -72,8 +72,10 @@
     
     void operator()(const FieldD &src, FieldD &psi) {
       LinearOperatorBase<FieldF> *Linop_f_use = &Linop_f;
-      bool using_fallback = false;
-      
+      bool using_fallback = true;
+      Linop_f_use = Linop_fallback;
+      std::cout << GridLogMessage << "ConjugateGradientReliableUpdate starting with fallback linear operator" << std::endl;
+
       psi.checkerboard = src.checkerboard;
       conformable(psi, src);
 
@@ -168,7 +170,7 @@
 	}
 	  
 	// Stopping condition
-	if (cp <= rsq) {
+	if (cp <= rsq || k==MaxIterations) {
 	  //Although not written in the paper, I assume that I have to add on the final solution
 	  precisionChange(mmp, psi_f);
 	  psi = psi + mmp;
@@ -182,7 +184,11 @@
 	  RealD resnorm = sqrt(norm2(p));
 	  RealD true_residual = resnorm / srcnorm;
 
-	  std::cout << GridLogMessage << "ConjugateGradientReliableUpdate Converged on iteration " << k << " after " << l << " reliable updates" << std::endl;
+	  if(cp<=rsq){
+	    std::cout << GridLogMessage << "ConjugateGradientReliableUpdate Converged on iteration " << k << " after " << l << " reliable updates" << std::endl;
+	  }else{
+	    std::cout << GridLogMessage << "ConjugateGradientReliableUpdate NOT Converged on iteration " << k << " after " << l << " reliable updates" << std::endl;
+	  }
 	  std::cout << GridLogMessage << "\tComputed residual " << sqrt(cp / ssq)<<std::endl;
 	  std::cout << GridLogMessage << "\tTrue residual " << true_residual<<std::endl;
 	  std::cout << GridLogMessage << "\tTarget " << Tolerance << std::endl;
@@ -194,7 +200,7 @@
 
 	  IterationsToComplete = k;	
 	  ReliableUpdatesPerformed = l;
-	  
+	  DoFinalCleanup = false; //--dsh
 	  if(DoFinalCleanup){
 	    //Do a final CG to cleanup
 	    std::cout << GridLogMessage << "ConjugateGradientReliableUpdate performing final cleanup.\n";
@@ -203,9 +209,9 @@
 	    CG(Linop_d,src,psi);
 	    IterationsToCleanup = CG.IterationsToComplete;
 	  }
-	  else if (ErrorOnNoConverge) assert(true_residual / Tolerance < 10000.0);
+	  //else if (ErrorOnNoConverge) assert(true_residual / Tolerance < 10000.0); //--dsh
 
-	  std::cout << GridLogMessage << "ConjugateGradientReliableUpdate complete.\n";
+	  std::cout << GridLogMessage << "ConjugateGradientReliableUpdate complete.  FIX THIS, DAN\n";
 	  return;
 	}
 	else if(cp < Delta * MaxResidSinceLastRelUp) { //reliable update
@@ -242,7 +248,7 @@
       std::cout << GridLogMessage << "ConjugateGradientReliableUpdate did NOT converge"
 		<< std::endl;
       
-      if (ErrorOnNoConverge) assert(0);
+      //if (ErrorOnNoConverge) assert(0); --dsh
       IterationsToComplete = k;
       ReliableUpdatesPerformed = l;      
     }    
