diff --git a/3.3.3.tar.bz2 b/3.3.3.tar.bz2
deleted file mode 100644
index 5b58f6fa..00000000
Binary files a/3.3.3.tar.bz2 and /dev/null differ
diff --git a/lib/algorithms/iterative/ConjugateGradientReliableUpdate.h b/lib/algorithms/iterative/ConjugateGradientReliableUpdate.h
index 13666f97..a6a847ad 100644
--- a/lib/algorithms/iterative/ConjugateGradientReliableUpdate.h
+++ b/lib/algorithms/iterative/ConjugateGradientReliableUpdate.h
@@ -194,7 +194,7 @@ namespace Grid {
 
 	  IterationsToComplete = k;	
 	  ReliableUpdatesPerformed = l;
-	  
+	  DoFinalCleanup = false; //--dsh
 	  if(DoFinalCleanup){
 	    //Do a final CG to cleanup
 	    std::cout << GridLogMessage << "ConjugateGradientReliableUpdate performing final cleanup.\n";
@@ -203,9 +203,9 @@ namespace Grid {
 	    CG(Linop_d,src,psi);
 	    IterationsToCleanup = CG.IterationsToComplete;
 	  }
-	  else if (ErrorOnNoConverge) assert(true_residual / Tolerance < 10000.0);
+	  //else if (ErrorOnNoConverge) assert(true_residual / Tolerance < 10000.0); //--dsh
 
-	  std::cout << GridLogMessage << "ConjugateGradientReliableUpdate complete.\n";
+	  std::cout << GridLogMessage << "ConjugateGradientReliableUpdate complete.  FIX THIS, DAN\n";
 	  return;
 	}
 	else if(cp < Delta * MaxResidSinceLastRelUp) { //reliable update
@@ -242,7 +242,7 @@ namespace Grid {
       std::cout << GridLogMessage << "ConjugateGradientReliableUpdate did NOT converge"
 		<< std::endl;
       
-      if (ErrorOnNoConverge) assert(0);
+      //if (ErrorOnNoConverge) assert(0); --dsh
       IterationsToComplete = k;
       ReliableUpdatesPerformed = l;      
     }    
